ledger/accountdb.go

```diff
@@ -982,27 +986,41 @@ func writeCatchpointStagingBalances(ctx context.Context, tx *sql.Tx, bals []norm
 	var rowID int64
 	for _, balance := range bals {
 		result, err = insertAcctStmt.ExecContext(ctx, balance.address[:], balance.normalizedBalance, balance.encodedAccountData)
-		if err != nil {
-			return err
-		}
-		aff, err := result.RowsAffected()
-		if err != nil {
-			return err
-		}
-		if aff != 1 {
-			return fmt.Errorf("number of affected record in insert was expected to be one, but was %d", aff)
-		}
-		rowID, err = result.LastInsertId()
-		if err != nil {
-			return err
+		if err == nil {
+			var aff int64
+			aff, err = result.RowsAffected()
+			if err != nil {
+				return err
+			}
+			if aff != 1 {
+				return fmt.Errorf("number of affected record in insert was expected to be one, but was %d", aff)
+			}
+			rowID, err = result.LastInsertId()
+			if err != nil {
+				return err
+			}
+		} else {
+			var sqliteErr sqlite3.Error
+			if errors.As(err, &sqliteErr) && sqliteErr.Code == sqlite3.ErrConstraint && sqliteErr.ExtendedCode == sqlite3.ErrConstraintUnique  {
+				// address exists: overflowed account record: find addrid
```

Comment:
didn't expect you would rely on the DB to tell you if you re-used the same account twice, I assumed you would just be able to tell as you were going through the `bals` list and saw a duplicate. But I guess you didn't want to assume the `bals` list is ordered?

Reply:
Maybe you could add a check here to make sure that the balance.encodedAccountData is empty?

Reply:
doesn't this mean you're on the second chunk for this account, and it's all resources now?

Reply:
the idea is base account info is always set, and overflowed entries just chunks with some more data.
Nicholas had a prev version with tracking account address from the top (see https://github.com/algorand/go-algorand/pull/4254/commits/d895c5d8e182401dd56b36b3308049ed9a02dccc) and this did not look elegant at all.
We do have resources counting logic so the applying catchpoint would break if Total* counters mismatch to actual counts (@nicholasguoalgorand do we have a test for this?) so relying on DB looks safe here.

Reply:
oh I see. so the chunks really could come in any order, then, too. OK! but only one of the encodedAccountData blobs would actually make it into the accounts table — hopefully they are all the same.